pkg/compact/compact.go

```diff
@@ -913,22 +913,33 @@ func (c *BucketCompactor) Compact(ctx context.Context) error {
 			return errors.Wrap(err, "build compaction groups")
 		}
 		finishedAllGroups := true
+		var wg sync.WaitGroup
+		errChan := make(chan error, len(groups))
 		for _, g := range groups {
-			shouldRerunGroup, _, err := g.Compact(ctx, c.compactDir, c.comp)
-			if err == nil {
-				if shouldRerunGroup {
-					finishedAllGroups = false
+			wg.Add(1)
+			go func(g *Group) {
```

Comment:
We need some workers-like design rather than this. Look this for comparison: https://github.com/improbable-eng/thanos/pull/887

Reply:
done